iconservice/fee/fee_storage.py

```diff
@@ -0,0 +1,135 @@
+
+if TYPE_CHECKING:
+    from ..database.db import ContextDatabase
+    from ..iconscore.icon_score_context import IconScoreContext
+
+
+class FeeStorage(object):
+    """Fee and Deposit state manager embedding a state db wrapper"""
+
+    _FEE_PREFIX = b'\x02'
+
+    def __init__(self, db: 'ContextDatabase') -> None:
+        """Constructor
+
+        :param db: (Database) state db wrapper
+        """
+        self._db = db
+
+    @property
+    def db(self) -> 'ContextDatabase':
+        """Returns state db wrapper.
+
+        :return: (Database) state db wrapper
+        """
+        return self._db
+
+    def _generate_key(self, key_data: bytes):
+        """
+        Generates a db key
+        """
+        return self._FEE_PREFIX + sha3_256(key_data).digest()
+
+    def get_deposit_meta(self, context: 'IconScoreContext', score_address: 'Address') -> 'DepositMeta':
+        """Returns the score deposit meta information.
+
+        :param context: Object that contains the useful information to process user's JSON-RPC request
+        :param score_address: SCORE address
+        :return: DepositMeta object
+        """
+        key = self._generate_key(score_address.to_bytes())
+        value = self._db.get(context, key)
+        return DepositMeta.from_bytes(value) if value else value
+
+    def put_deposit_meta(self, context: 'IconScoreContext', score_address: 'Address',
+                         deposit_meta: 'DepositMeta') -> None:
+        """Puts the score deposit meta information into db.
+
+        :param context: Object that contains the useful information to process user's JSON-RPC request
+        :param score_address: SCORE address
+        :param deposit_meta: DepositMeta object
+        :return: None
+        """
+        key = self._generate_key(score_address.to_bytes())
+        value = deposit_meta.to_bytes()
+        self._db.put(context, key, value)
+
+    def delete_deposit_meta(self, context: 'IconScoreContext', score_address: 'Address') -> None:
+        """Deletes the score deposit meta information from db.
+
+        :param context: Object that contains the useful information to process user's JSON-RPC request
+        :param score_address: SCORE address
+        :return: None
+        """
+        key = self._generate_key(score_address.to_bytes())
+        self._db.delete(context, key)
+
+    def get_deposit(self, context: 'IconScoreContext', deposit_id: bytes) -> 'Deposit':
+        """Returns the deposit.
+
+        :param context: Object that contains the useful information to process user's JSON-RPC request
+        :param deposit_id: Deposit id
+        :return: Deposit Object
+        """
+        key = self._generate_key(deposit_id)
+        value = self._db.get(context, key)
+
+        if value:
+            value = Deposit.from_bytes(value)
+            value.id = deposit_id
+
+        return value
+
+    def put_deposit(self, context: 'IconScoreContext', deposit_id: bytes, deposit: 'Deposit') -> None:
```

Comment:
```suggestion
    def put_deposit(self, context: 'IconScoreContext', deposit: 'Deposit') -> None:
```
The `deposit_id` param is redundant. The id can be retrieved from Deposit object itself.